docs/source/developers/components_exports.rst

```diff
@@ -0,0 +1,483 @@
+Components and component members
+================================
+
+Components are, simply put, subapplications. They have their own templates,
+satic assets, route handlers, database migrations, and so on. Each of these
+things are called 'component members'. Some members may require load ordering
+(e.g., plugins may depend on other plugins, some routes may require to be
+registered after others or before others). We call these ordering requirements
+'component member dependencies'.
+
+Component member groups
+-----------------------
+
+Component member groups (CMGs) are groups of component members by type. We
+currently have the following groups:
+
+- databases
+- templates
+- static assets
+- hooks
+- plugins
+- WSGI middleware
+- route handlers
+- command line argument handlers
+
+Each component may have one or more members that belong to one of the CMGs. The
+CMGs are also known as component member type. 
+
+All members from all components that belong to the same group comprise that
```

Comment:
`All members from all components that belong to the same group comprise that group` is a bit hard to parse.